### ISSUE TITLE
How long does it take for the training process?

### ISSUE THREAD

**Issue by shizhediao** [allenai/dont-stop-pretraining#16]
Hi,
I am doing DAPT on CS domain with 38 GB CS data on a single TPU V3-8.
It is estimated that will cost 20-24 hours for one epoch.
I see from the paper you use TPU V3-8 as well but I do not find the time information in the paper. Would you like to share how much time you need for pretraining?
Thanks!

**Comment by kyleclo** [comment]
Hi @shizhediao it depends a bit on your implementation & what you're trying to do.  We were using an early beta version of torch-xla and one of the TPUs was a bit unstable, so wall clock time estimates are longer than what you're probably going to see. 

For reference, we sharded the pretraining corpus into separate files.  Each file took roughly 23~26 min to finish performing 55 gradient updates (at effective batch size of 2048 through use of gradient accumulation).  We did this for 12.5K updates (i.e. finished 227 shards) before terminating the experiment, so roughly 5675 min (94 hrs or 4 days).
